cmd/promtool/main.go

```diff
@@ -194,54 +196,56 @@ func checkConfig(filename string) ([]string, error) {
 	}
 
 	var ruleFiles []string
-	for _, rf := range cfg.RuleFiles {
-		rfs, err := filepath.Glob(rf)
-		if err != nil {
-			return nil, err
-		}
-		// If an explicit file was given, error if it is not accessible.
-		if !strings.Contains(rf, "*") {
-			if len(rfs) == 0 {
-				return nil, errors.Errorf("%q does not point to an existing file", rf)
+	// Return empty slice of rules and don't check certs/tokens if syntax-only check specified
+	if !syntaxOnly {
+		for _, rf := range cfg.RuleFiles {
+			rfs, err := filepath.Glob(rf)
+			if err != nil {
+				return nil, err
 			}
-			if err := checkFileExists(rfs[0]); err != nil {
-				return nil, errors.Wrapf(err, "error checking rule file %q", rfs[0])
+			// If an explicit file was given, error if it is not accessible.
+			if !strings.Contains(rf, "*") {
+				if len(rfs) == 0 {
+					return nil, errors.Errorf("%q does not point to an existing file", rf)
+				}
+				if err := checkFileExists(rfs[0]); err != nil {
+					return nil, errors.Wrapf(err, "error checking rule file %q", rfs[0])
+				}
 			}
-		}
-		ruleFiles = append(ruleFiles, rfs...)
-	}
-
-	for _, scfg := range cfg.ScrapeConfigs {
-		if err := checkFileExists(scfg.HTTPClientConfig.BearerTokenFile); err != nil {
-			return nil, errors.Wrapf(err, "error checking bearer token file %q", scfg.HTTPClientConfig.BearerTokenFile)
+			ruleFiles = append(ruleFiles, rfs...)
 		}
 
-		if err := checkTLSConfig(scfg.HTTPClientConfig.TLSConfig); err != nil {
-			return nil, err
-		}
+		for _, scfg := range cfg.ScrapeConfigs {
+			if err := checkFileExists(scfg.HTTPClientConfig.BearerTokenFile); err != nil {
```

Comment:
This isn't a full list of the files.

Reply:
I have not modified this section of code, other than wrapping it inside the `if !syntaxOnly {...}` block

Reply:
It's still something that needs fixing, and is related.

Reply:
What's it missing?

Reply:
I make it consul openstack marathon triton alertmanager remote_write remote_read

The current approach clearly isn't very maintainable. Something smarter would be good.